nifi-nar-bundles/nifi-provenance-repository-bundle/nifi-persistent-provenance-repository/src/main/java/org/apache/nifi/provenance/lucene/DocsReader.java

```diff
@@ -100,101 +96,61 @@ private ProvenanceEventRecord getRecord(final Document d, final RecordReader rea
             }
         }
 
-        if ( record == null ) {
-            throw new IOException("Failed to find Provenance Event " + d);
-        } else {
-            return record;
+        if (record == null) {
+            logger.warn("Failed to read Provenance Event for '" + d + "'. The event file may be missing or corrupted");
         }
-    }
 
+        return record;
+    }
 
     public Set<ProvenanceEventRecord> read(final List<Document> docs, final Collection<Path> allProvenanceLogFiles,
-        final AtomicInteger retrievalCount, final int maxResults, final int maxAttributeChars) throws IOException {
-        if (retrievalCount.get() >= maxResults) {
-            return Collections.emptySet();
-        }
-
-        LuceneUtil.sortDocsForRetrieval(docs);
```

Comment:
Hmm, I don't think so. Based on my observation it was done to group files based on storage file name and sorting had that sideeffect, so in a way the additional sorting is a bit of an overkill here, hence refactoring and new utility operation.